dump_dbus_util.hpp

```diff
@@ -37,4 +43,49 @@ uint64_t getDumpSize(sdbusplus::bus::bus& bus, const std::string& objectPath);
  * @return D-Bus entries with properties
  */
 ManagedObjectType getDumpEntries(sdbusplus::bus::bus& bus);
+
+/**
+ * @brief Read D-Bus property to check if system is HMC managed
+ * @param[in] bus D-Bus handle
+ * @return true if HMC managed else false
+ */
+bool isSystemHMCManaged(sdbusplus::bus::bus& bus);
+
+/**
+ * @brief Read property value from the specified object and interface
+ * @param[in] bus D-Bus handle
+ * @param[in] service service which has implemented the interface
+ * @param[in] object object having has implemented the interface
+ * @param[in] intf interface having the property
+ * @param[in] prop name of the property to read
+ * @return property value
+ */
+template <typename T>
+T readDBusProperty(sdbusplus::bus::bus& bus, const std::string& service,
+                   const std::string& object, const std::string& intf,
+                   const std::string& prop)
+{
+    T retVal{};
+    try
+    {
+        auto properties =
+            bus.new_method_call(service.c_str(), object.c_str(),
+                                "org.freedesktop.DBus.Properties", "Get");
+        properties.append(intf);
+        properties.append(prop);
+        auto result = bus.call(properties);
+        result.read(retVal);
+    }
+    catch (const std::exception& ex)
+    {
+        log<level::ERR>(
+            fmt::format("Failed to get the property ({}) interface ({}) "
+                        "error ({}) ",
+                        prop.c_str(), intf.c_str(), ex.what())
+                .c_str());
+        throw;
```

Comment:
if we cant read the bios attribute what should be the default behavior?

Reply:
Now modified to check if the attribute is not set of the BIOS object is not created we willl assume that system is not HMC managed.